addons/stock/models/stock_move.py

```diff
@@ -1022,8 +1028,10 @@ def _action_done(self):
         for move in moves:
             if move.state == 'cancel' or move.quantity_done <= 0:
                 continue
-            moves_todo |= move
             moves_todo |= move._create_extra_move()
+            # extra move will not be merged in mrp thus we
+            # have to process both moves.
+            moves_todo |= move.exists()
```

Comment:
I would like to remove it however extra moves are not merged if move doesn't have a picking_id which explain the code that split move line in create extra move.